lib/graph3d/Graph3d.js

```diff
@@ -1680,6 +1646,40 @@ Graph3d.prototype._getColorsSize = function() {
 };
 
 
+/**
+ * Determine the color corresponding to a given value on the color scale.
+ *
+ * @param {number} [x] the data value to be mapped running from 0 to 1
+ * @param {number} [v] scale factor between 0 and 1 for the color brightness
+ * @returns {string}
+ * @private
+ */
+Graph3d.prototype._colormap = function(x, v) {
+  if (v === undefined) {
+    v = 1;
+  }
+
+  var colormap = this.colormap;
+  if (Array.isArray(colormap)) {
+    var maxIndex = colormap.length - 1;
+    var startIndex = Math.max(Math.floor(x * maxIndex), 0);
+    var endIndex = Math.min(startIndex + 1, maxIndex);
+    var innerRatio = x * maxIndex - startIndex;
+    var min = colormap[startIndex];
+    var max = colormap[endIndex];
+    var r = min.r + innerRatio*(max.r - min.r);
+    var g = min.g + innerRatio*(max.g - min.g);
+    var b = min.b + innerRatio*(max.b - min.b);
```

Comment:
Could you use let and const instead of var? Especially here (but don't use var anywhere, please) it's weird and confusing because you "declare" them in a block but in reality they leak all over the function.

Reply:
Agreed that's cleaner. Again, these variables were declared as `var` before I touched the code, so I just copied this over, but it's probably better to do some cleanup here. I'm not all too familiar with JS (especially not ES>=6 features) so I will overlook these kinds of issues. 

Except for `r, g, b`: The code sets them differently within each `if-else` branch to use them after the block, so `const` is out of the picture and we need function scope anyway. Now, the object destructuring `var {r, g, b} = colormap(x);` in the block below needs additional parentheses if not combined with a declaration, which looks weird.